src/ds/champ_map.h

```diff
@@ -103,6 +104,14 @@ namespace champ
     return padding;
   }
 
+  template <class K, class V>
+  uint32_t static get_size_with_padding(const K& k, const V& v)
```

Comment:
We primarily use `size_t` for the sizes, but `uint32_t` here - what do we actually write?

Minor: For safety, we could add a `using SerializedSizeT = size_t;` in `champ_map_serializers`, and then we can distinguish uses in this file from _other_ `size_t`s (and change to `uint32_t` at a single point), but this is almost certainly overkill!